mud-contracts/world/test/smart-gate/SmartGateTest.t.sol

```diff
@@ -194,6 +195,76 @@ contract SmartGateTest is MudTest {
     assert(!smartGate.isGateLinked(sourceGateId, destinationGateId));
   }
 
+  function testLinkUnLinkMultipleSmartGates() public {
+    uint256 smartGateA = 123;
+    uint256 smartGateB = 124;
+    uint256 smartGateC = 125;
+    uint256 smartGateD = 126;
+
+    testAnchorSmartGate(smartGateA);
+    testAnchorSmartGate(smartGateB);
+    testAnchorSmartGate(smartGateC);
+    testAnchorSmartGate(smartGateD);
+
+    //link AB
+    smartGate.linkSmartGates(smartGateA, smartGateB);
+    SmartGateLinkTableData memory linkDataAB = SmartGateLinkTable.get(smartGateA);
+    assertEq(linkDataAB.destinationGateId, smartGateB);
+    assertTrue(linkDataAB.isLinked);
+
+    SmartGateLinkTableData memory linkDataBA = SmartGateLinkTable.get(smartGateB);
+    assertEq(linkDataBA.destinationGateId, smartGateA);
+    assertTrue(linkDataBA.isLinked);
+
+    //revert when linking BA because B and A its already linked
+    vm.expectRevert(
+      abi.encodeWithSelector(SmartGateSystem.SmartGate_GateAlreadyLinked.selector, smartGateB, smartGateA)
+    );
+    smartGate.linkSmartGates(smartGateB, smartGateA);
+
+    //revert when link BC because B its already linked
+    vm.expectRevert(
+      abi.encodeWithSelector(SmartGateSystem.SmartGate_GateAlreadyLinked.selector, smartGateB, smartGateC)
+    );
+    smartGate.linkSmartGates(smartGateB, smartGateC);
```

Comment:
we should allow A->B, B->C link right?

Reply:
Yes I thought about this but If we allow destination then it will break, and this is what happened in that video

A -> B  then we have AB true , BA true
allowing destination
B > C  then we have BC true, CB is true

When we unlink BC, in table it finds all references for B and makes isLinked `false`
BC false 
BA false 

because table has one primary key.

Reply:
So they have to first link A -> B jump and then unlink AB then link BC then jump

Reply:
@vayan for now it is just A <-> B and then C <-> D